src/cluster_legacy.h

```diff
@@ -90,7 +92,12 @@ typedef struct clusterNodeFailReport {
 #define CLUSTERMSG_TYPE_MFSTART 8               /* Pause clients for manual failover */
 #define CLUSTERMSG_TYPE_MODULE 9                /* Module cluster API message. */
 #define CLUSTERMSG_TYPE_PUBLISHSHARD 10         /* Pub/Sub Publish shard propagation */
-#define CLUSTERMSG_TYPE_COUNT 11                /* Total number of message types. */
+#define CLUSTERMSG_TYPE_PUBLISH_LIGHT 11        /* Pub/Sub Publish propagation using light header*/
+#define CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT 12   /* Pub/Sub Publish shard propagation using light header*/
+#define CLUSTERMSG_TYPE_COUNT 13                /* Total number of message types. */
+
+#define IS_PUBSUB_LIGHT_MESSAGE(type)                                                                                  \
+    (type == CLUSTERMSG_TYPE_PUBLISH_LIGHT || type == CLUSTERMSG_TYPE_PUBLISHSHARD_LIGHT)
```

Comment:
This macro can get long in the future if we start adding more message types. 

I would highly recommend having a single bit called IS_LIGHT_MESSAGE and move the message type to a different field.

Reply:
The counterpoint to this is we would have mixed usage of the `type` field going forward. Currently, it signifies the `data type` as you call it and not the header type. 

This would lead to having different understanding for each header type i.e. for `clusterMsg` and `clusterLightMsg` and have more conditional statement to handle each of it. I think we can stick with the existing approach and keep it generic for both.

Reply:
I see. The introduction of this `data_type` field does make the existing code more complex. 

I like @zuiderkwast's [idea](https://github.com/valkey-io/valkey/pull/654#discussion_r1683894061) (with a twist). I think it can achieve the same effect as I was looking for.

Reply:
Yeah, that works. 👍